### ISSUE TITLE
Proposal: Please start using Semantic Versioning

### ISSUE THREAD

**Issue by KateGo520** [etherlabsio/healthcheck#26]
I found that this project already [supports Go modules](https://github.com/etherlabsio/healthcheck/blob/master/go.mod). But sadly, the [tags](https://github.com/etherlabsio/healthcheck/tags) doesn't follow [Semantic Versioning](https://semver.org), which means that all tags of this project will be ignored by Go modules and replaced by pseudo-versions, go get acts weirdly when tags are not in that form. It would be great to have the tagged release be named in the format vX.X.X format so that go mod can read it. 
```
	github.com/etherlabsio/healthcheck v0.0.0-20191224061800-dd3d2fd8c3f6
```
Else the mod file shows something like `github.com/etherlabsio/healthcheck v0.0.0-20191224061800-dd3d2fd8c3f6 ` which is not very readable and difficult to upgrade. It’s hard to verify which version is in use. This is not conducive to version control.

So, I propose this project to follow Semantic Versioning in future versions. For example, `v1.0.1`, `v2.0.0`, `v3.1.0-alpha`, `v3.1.0-beta.2`etc, so that other project can use tag in go.mod.

**Comment by KateGo520** [comment]
@vthiery  @karthikmuralidharan  Could you help me review this issue? Thx :p

**Comment by vthiery** [comment]
@KateGo520  I'm not maintainer of this repository but I can bring some elements to answer your request:

For consistency with existing tags (> 2.x.x), I guess the next version should be v2.x.x (assuming @karthikmuralidharan is willing to adopt semver). I believe this would put us into the Go module v2 realm and require the following patch:

```git
diff --git a/go.mod b/go.mod
index 324e5a3..20ae9a1 100644
--- a/go.mod
+++ b/go.mod
@@ -1,3 +1,3 @@
-module github.com/etherlabsio/healthcheck
+module github.com/etherlabsio/healthcheck/v2

 go 1.12
```

It would also require users to change the path on their side, should they upgrade.

**Comment by karthikmuralidharan** [comment]
Hi @KateGo520 

Moving to proper semvar makes total sense to me. I also agree with @vthiery 's assessment that we should be adopting v2.x.x as the next version fully compatible with Go modules.

I'm open to a contribution if you're interested. 
Thanks

**Comment by KateGo520** [comment]
@karthikmuralidharan   Thank you for your reply. I totally agree with @vthiery 's assessment . That's the right way to use Go Modules.

**Comment by llorllale** [comment]
> Please start using Semantic Versioning

This project already is. See the [BNF](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions). 

From what I understand the "v" prefix is just baggage projects keep dragging around from the old days.

**Comment by SuNNjek** [comment]
@llorllale Not according to https://blog.golang.org/publishing-go-modules (which I would say is a better source on how to version Go modules). The "v" seems to be necessary actually.

<blockquote><div><strong><a href="https://blog.golang.org/publishing-go-modules">Publishing Go Modules - The Go Blog
Search</a></strong></div><div>Go is an open source programming language that makes it easy to build simple, reliable, and efficient software.</div></blockquote>

**Comment by llorllale** [comment]
@SuNNjek 

> @llorllale Not according to https://blog.golang.org/publishing-go-modules (which I would say is a better source on how to version Go modules). The "v" seems to be necessary actually.
> 
> > **[Publishing Go Modules - The Go Blog
> > Search](https://blog.golang.org/publishing-go-modules)**Go is an open source programming language that makes it easy to build simple, reliable, and efficient software.

Yes - the golang blog is a better source on how to version Go modules.

I was implying that the Go module designers may simply have had some preference for the "v" prefix (ie. "baggage") but I haven't looked into it.

This project is already following the semver spec but since it's a Go module it also has to adhere to the Go tooling. You don't have this problem in other dependency managers such as Maven.

**Comment by llorllale** [comment]
TLDR I was being pedantic. Sorry.
